app/plugins/eu.esdihumboldt.hale.app.transform/src/eu/esdihumboldt/hale/app/transform/ExecTransformation.java

```diff
@@ -457,19 +438,89 @@ private void transform() throws InterruptedException, ExecutionException {
 		ListenableFuture<Boolean> res = Transformation.transform(sources, target, env,
 				reportHandler, id, validators, context.getFilters());
 
-		if (res.get()) {
+		boolean orgSuccess = res.get();
+
+		// Job threads might still be active, wait a moment to allow
+		// them to complete and file their report (otherwise error may get lost)
+		try {
+			Thread.sleep(3000);
+		} catch (Throwable e) {
+			// ignore
+		}
+
+		boolean success;
+		try {
+			success = evaluateSuccess(orgSuccess);
+		} catch (Throwable e) {
+			error("Success evaluation resulted in an error:\n" + e.getMessage());
+			if (context.isLogException()) {
+				e.printStackTrace();
+			}
+			return 2;
+		}
+
+		if (success) {
 			info("Transformation completed. Please check the reports for more details.");
 		}
 		else {
-			// Job threads might still be active, wait a moment to allow them to
-			// complete and file their report (otherwise error may get lost)
-			try {
-				Thread.sleep(3000);
-			} catch (Throwable e) {
-				// ignore
+			if (orgSuccess) {
+				error("Transformation failed according to the success evaluation script.");
+				return 2;
+			}
+			else {
+				error("Transformation failed, please check the reports for details.");
+				return 1;
+			}
+		}
+
+		// exit OK
+		return 0;
+	}
+
+	private boolean evaluateSuccess(boolean success) {
+		StatsCollector statistics = reportHandler.getStatistics();
+
+		if (context.getStatisticsFile() != null) {
+			// write to statistics file
+			try (Writer writer = Files.newBufferedWriter(context.getStatisticsFile().toPath(),
+					StandardCharsets.UTF_8)) {
+				writer.write(JsonOutput.prettyPrint(statistics.saveToJson(true)));
+			} catch (IOException e) {
+				ExecUtil.error("Error writing statistics file: " + e.getMessage());
+				// XXX should this fail the command?
```

Comment:
I think it's ok not fail here b/c the evaluation script will fail anyway as soon as it tries an `assert` on one of the statistics. And if the stats file isn't used (e.g. in the edge cases of an empty evaluation script or one that doesn't use any of the statistics) there's no reason to fail the command.

Reply:
There could be a call that just outputs the statistics, w/o using an evaluation script. But I would still leave it as it is right now.